app/views/posts/show.html.erb

```diff
@@ -21,10 +21,17 @@
 <% if @post.comments.count > 0 %>
   <div class='post_comments'>
     <% @post.comments.each do |comment| %>
-    <p class='post_comment'>
-      <%= comment.author.name%>:
-      <%= comment.body %>
-    </p>
+    <div class='comment_container'>
+      <p class='post_comment'>
+        <%= comment.author.name%>:
+        <%= comment.body %>
+      </p>
+      <p class='delete_btn'>
+        <% if user_signed_in? &&  (current_user.role == 'admin' || current_user == comment.author) %>
```

Comment:
- [ ] We can also improve this conditional here and use the `cancancan` way. Instead of this we can use this code.

  ```
  <% if user_signed_in? && (can? :destroy, post) %>
  ```